### ISSUE TITLE
Setting an empty value fails with MODULE FAILURE

### ISSUE THREAD

**Issue by mss** [cmprescott/ansible-xml#63]
This might be the same issue as #53 and maybe #7.

The XML:

``` xml
<?xml version='1.0' encoding='UTF-8'?>
<localconfig>
  <key name="smtp_notify">
    <value>yes</value>
  </key>
</localconfig>
```

This task works to set the value above:

``` yaml
  - name: Set localconfig options
    xml:
      file: "/opt/zimbra/conf/localconfig.xml"
      xpath: "/localconfig/key[@name='smtp_notify']/value"
      value: "yes"
```

But this fails:

``` yaml
  - name: Set localconfig options
    xml:
      file: "/opt/zimbra/conf/localconfig.xml"
      xpath: "/localconfig/key[@name='smtp_notify']/value"
      value: ""
```

Expected result:

``` xml
<?xml version='1.0' encoding='UTF-8'?>
<localconfig>
  <key name="smtp_notify">
    <value></value>
  </key>
</localconfig>
```

Actual result:

```
TASK [Set localconfig options] *************************************************
fatal: [zmdir2.example.com]: FAILED! => {"changed": false, "failed": true, "module_stderr": "", "module_stdout": "", "msg": "MODULE FAILURE", "parsed": false}
```

This is version 0.3.1 if I'm not mistaken (whatever was current on galaxy today).

**Comment by mss** [comment]
Hmm… was this maybe fixed with #60?

**Comment by pigulla** [comment]
This is still broken :-(

**Comment by dagwieers** [comment]
The xml module is upstream now (and ships with Ansible v2.4).

**Please close this issue**, and retest against the new upstream xml module. If needed, open a new issue at: https://github.com/ansible/ansible/issues

**Comment by mss** [comment]
This is fixed in the version which is shipped with Ansible v2.4.

Tested with the XML file from the initial report and following playbook:
```yaml
---
- hosts: all
  tasks:
  - name: Set localconfig options
    xml:
      file: "/tmp/localconfig.xml"
      xpath: "/localconfig/key[@name='smtp_notify']/value"
      value: ""
```

**Comment by dagwieers** [comment]
Thanks for reporting back !
